pages/pokedex/index.tsx

```diff
@@ -1,24 +1,50 @@
-import type { InferGetServerSidePropsType, NextPage } from "next";
-import { CardGrid } from "../../components";
-import styles from "../../styles/Page.module.css";
+import { gql } from '@apollo/client';
+import type { Pokemon } from '..';
+import { getApolloClient } from '../../apollo-client';
+import type { InferGetServerSidePropsType, NextPage } from 'next';
+import { CardGrid } from '../../components';
+import styles from '../../styles/Page.module.css';
+import { PokemonCard } from '../../components';
+import { POKEMON } from '../../config';
 
 export const getServerSideProps = async () => {
+  const client = getApolloClient();
+
+  const { data } = await client.query({
```

Comment:
You'll want to type the response here, much like I did in the index page

Reply:
what response? the query itself? I forgot to ask about the document head... it's returned by getServerSideProps but not used in the homepage? I can see this being useful for a per-page for each individual pokemon though having their own "title" (for the page)

Reply:
Similar to what's happening here: https://github.com/brightsize/pokemon-apollo/blob/main/pages/index.tsx#L23

The `client.query` method uses [Type Generics](https://www.typescriptlang.org/docs/handbook/2/generics.html) to allow you to specify the type that will be returned from the call.

That way (assuming the GraphQL Query is correct, and the network request doesn't fail) we can use TypeScript to guarantee what our return value is - rather than relying on `any` or `unknown`

Reply:
Nice. That type generics doc page really filled in some gaps for me.